File: Day-09: Recursion/6) K-th permutation Sequence.java
class Solution {
    public String getPermutation(int n, int k) {
        List<Integer> nums = new ArrayList<>();
        for(int i = 1; i <= n; i++)
        {
            nums.add(i);
        }
        int[] fact = new int[n];
        fact[0] = 1;
        for(int i = 1; i < n; i++)
        {
            fact[i] = fact[i - 1] * i;
        }
        k--;
        StringBuilder ans = new StringBuilder();
        for(int i = 0; i < n; i++)
        {
            int id = k / fact[n - i - 1];
            ans.append(nums.get(id));
            nums.remove(id);
            k %= fact[n - i - 1];
        }
        return ans.toString();
    }
}

File: Day-09: Recursion/6) K-th permutation Sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> nums;
        for(int i = 1; i <= n; i++)
        {
            nums.push_back(i);
        }
        string ans = "";
        vector<int> fact(n, 1);
        for(int i = 1; i < n; i++)
        {
            fact[i] = fact[i-1] * i;
        }
        k--;
        for(int i = 0; i < n; i++)
        {
            int id = k / fact[n-i-1];
            ans += to_string(nums[id]);
            nums.erase(nums.begin() + id);
            k %= fact[n-i-1];
        }
        return ans;
    }
};
